Install Oracle Instant Client

calumabarnett opened the pull request:
This PR will install Oracle Instant Client (19.6.0.0.0). This will enable analysts to directly query Oracle databases using the `cx_Oracle` package. This is in support of an interim measure to provide access to the Delius database on the Analytical Platform while work on the pipeline is ongoing.

xoen commented on datascience-notebook/Dockerfile:
```diff
@@ -10,11 +10,25 @@ ENV CHOWN_HOME=no
 COPY ./files/apt_packages /tmp/
 
-RUN conda install boto3 black flake8 jupyterlab_code_formatter \
+RUN mkdir /opt/oracle \
```
I'd add a comment above this command to explain what's installing, etc...some clues for the future us (or them).

davidread commented:
@calumabarnett We talked about making this image with the Oracle client available only to limited users, who are manually given this image - see https://docs.google.com/document/d/1ISoxf47qF3l0VqBo3d-yY4SekCQoBmmGiqlnVowtoaE/edit#
How about we achieve this in a similar way to the all-spark image, by putting the Dockerfile in a separate directory in this repo? It just needs someone to setup Quay to build it and then it's available for users to specify it in their Deployment.

calumabarnett commented:
Thanks @davidread – sounds like a sensible approach. I've reverted the `datascience-notebook` Dockerfile, moved the updated Dockerfile to a new directory called `oracle-datascience-notebook` and created a new Quay repo of the same name to build from this directory.

davidread commented:
That build error about missing `[jobs]` suggests the syntax in the https://github.com/ministryofjustice/analytics-platform-jupyter-notebook/blob/master/circle.yml needs a little update, if you can work that out too before merge?

xoen commented:
Actually the comment about the repetition as I wonder whether this `Dockerfile` could use the standard JupyterLab image as `BASE` and install this on top of that to make it clear it's the only difference.

@calhutchison sorry I didn't think/suggest about this before 😬 

Basically it's similar to what they do in the official R/Jupyter images where they have few common/base images and they have more specialised images which add/inherit on top of these images to add functionalities.

It makes a lot of sense as it clear what's special about this docker image as its `Dockerfile` would contain only the difference.

Just a thought.

calumabarnett commented:
> That build error about missing `[jobs]` suggests the syntax in the https://github.com/ministryofjustice/analytics-platform-jupyter-notebook/blob/master/circle.yml needs a little update, if you can work that out too before merge?

@davidread It's a very old v1 Circle config. I'm not sure it's used/needed anymore anyway as all PRs trigger a build on Quay directly. Perhaps what's needed is just to remove it?

davidread commented:
@calumabarnett Makes sense. I've removed the Circle config in this PR: https://github.com/ministryofjustice/analytics-platform-jupyter-notebook/pull/31

davidread commented:
This is good to merge now I think

calumabarnett commented:
@xoen that does make sense – I've quickly tried to build from the base image but have run into some errors with apt. I'm sure they could be fixed but given we have something working and I don't think the benefits are huge, I'm inclined to stick with things as they are.
